### ISSUE TITLE
Released v0.9.0 breaks Iris tests

### ISSUE THREAD

**Issue by ajdawson** [SciTools/biggus#135]
Iris testing on Travis CI appears to be broken after v0.9.0 was packaged:

```
======================================================================
ERROR: test_deferred_bytes (iris.tests.unit.fileformats.pp.test__create_field_data.Test__create_field_data)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Iris-1.8.0.dev0-py2.7-linux-x86_64.egg/iris/tests/unit/fileformats/pp/test__create_field_data.py", line 74, in test_deferred_bytes
    self.assertEqual(field._data.shape, data_shape)
  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Biggus-0.9.0-py2.7.egg/biggus/__init__.py", line 1018, in shape
    return _sliced_shape(self.concrete.shape, self._keys)
  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Biggus-0.9.0-py2.7.egg/biggus/__init__.py", line 2583, in _sliced_shape
    size = len(range(*key.indices(shape[shape_dim])))
TypeError: '_SentinelObject' object cannot be interpreted as an index
======================================================================
ERROR: test_real_data_cube_indexing (iris.tests.test_cdm.TestDataManagerIndexing)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Iris-1.8.0.dev0-py2.7-linux-x86_64.egg/iris/tests/test_cdm.py", line 972, in test_real_data_cube_indexing
    cube = self.cube[(0, 4, 5, 2), 0, 0]
  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Iris-1.8.0.dev0-py2.7-linux-x86_64.egg/iris/cube.py", line 1929, in __getitem__
    data = data[other_slice]
  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Biggus-0.9.0-py2.7.egg/biggus/__init__.py", line 525, in __getitem__
    keys = _full_keys(keys, self.ndim)
  File "/home/travis/miniconda/envs/test-environment/lib/python2.7/site-packages/Biggus-0.9.0-py2.7.egg/biggus/__init__.py", line 2622, in _full_keys
    if n_keys_non_newaxis - 1 >= ndim and Ellipsis in keys:
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

**Comment by pelson** [comment]
Looks reasonable to raise this here - I suspect it is a biggus bug of my doing. I'll take a look now.

**Comment by ajdawson** [comment]
Yeah I was 50/50 here or iris, but since the same people work on both I thought it would be better off here. Sorry for the vagueness of the actual report!

**Comment by pelson** [comment]
The second one is a biggus bug. I need to figure out a better way of checking if Ellipsis is in the keys by the looks of it:

```
>>> a = np.array([1, 3])
>>> keys = (a, Ellipsis)
>>> len([key for key in keys if key is not np.newaxis])
2
>>> Ellipsis in keys
Traceback (most recent call last):
  File "<stdin>", line 1, in <module>
ValueError: The truth value of an array with more than one element is ambiguous. Use a.any() or a.all()
```

**Comment by pelson** [comment]
Second item fixed by #136.

**Comment by pelson** [comment]
Fixed by v0.9.1. We've also just cut v0.10.0 which will also have the fix.
